src/reporting/clients/IMappingsClient.ts

```diff
@@ -0,0 +1,537 @@
+   * @param {string} iModelId Id of the source Mapping&#x27;s iModel.
+   * @param {string} mappingId Id of the source Mapping.
+   * @param {string} accessToken OAuth access token with scope `insights:modify`
+   * @param {MappingCopy} mappingCopy Request body.
+   * @memberof ReportingClient
+   * @link https://developer.bentley.com/apis/insights/operations/copy-mapping/
+   */
+  copyMapping(accessToken: AccessToken,
+    iModelId: string,
+    mappingId: string,
+    mappingCopy: MappingCopy
+  ): Promise<Mapping>;
+
+  /**
+   * Gets all Groups for a Mapping. This method returns the full list of groups.
+   * @param {string} iModelId The iModel Id.
+   * @param {string} mappingId The Mapping Id.
+   * @param {string} accessToken OAuth access token with scope `insights:read`
+   * @param {number} top The number of entities to load.
+   * @memberof ReportingClient
+   * @link https://developer.bentley.com/apis/insights/operations/get-groups/
+   */
+  getGroups(accessToken: AccessToken,
+    iModelId: string,
+    mappingId: string,
+    top?: number
+  ): Promise<Group[]>;
+
+  /**
+   * Gets an async paged iterator of Groups for a Mapping.
+   * This method returns an iterator which loads pages of groups as it is being iterated over.
+   * @param {string} iModelId The iModel Id.
+   * @param {string} mappingId The Mapping Id.
+   * @param {string} accessToken OAuth access token with scope `insights:read`
+   * @param {number} top The number of entities to load.
+   * @memberof ReportingClient
+   * @link https://developer.bentley.com/apis/insights/operations/get-groups/
+   */
+  getGroupsIterator(accessToken: AccessToken,
+    iModelId: string,
+    mappingId: string,
+    top?: number
+  ): EntityListIterator<Group>;
+
+  /**
+   * Creates a Group for an iModel data source Mapping.
+   * @param {string} iModelId The iModel Id.
+   * @param {string} mappingId Id of the Mapping for which to create a new Group.
+   * @param {string} AccessToken OAuth access token with scope `insights:modify`
+   * @param {GroupCreate} group Request body.
+   * @memberof ReportingClient
+   * @link https://developer.bentley.com/apis/insights/operations/create-group/
+   */
+  createGroup(accessToken: AccessToken,
+    iModelId: string,
+    mappingId: string,
+    group: GroupCreate
+  ): Promise<Group>;
+
+  /**
+   * Gets a Group for a Mapping.
+   * @param {string} iModelId The iModel Id.
+   * @param {string} mappingId The Mapping Id.
+   * @param {string} groupId The Group Id.
+   * @param {string} accessToken OAuth access token with scope `insights:read`
+   * @memberof ReportingClient
+   * @link https://developer.bentley.com/apis/insights/operations/get-group/
+   */
+  getGroup(accessToken: AccessToken,
+    iModelId: string,
+    mappingId: string,
+    groupId: string
+  ): Promise<Group>;
+
+  /**
+   * Updates a Group for a Mapping.
+   * @param {string} iModelId Globally Unique Identifier of the target iModel.
```

Comment:
Inconsistent documentation. Meaning of iModelId, mappingId, and groupId are the same in almost all urls. Change the documentation of these parameters to match `getGroup` documentation. I think it should only be different for the `copyMapping` operation, because the url parameters are for the source mapping.

Reply:
changed all documentation to match getGroup with the exception of copyMapping